### ISSUE TITLE
paddle v2  autoencoder data input

### ISSUE THREAD

**Issue by adrianhust** [PaddlePaddle/Paddle#9061]
Use paddle v2 to implement simple auto encoder;
Input: like this, [0,1,1,1,0,0,0], a vector with binary value;
then feed input to fc layer;

code:
```
def cluster_reader_cluster(base_dir, node_id):
    def data_reader():
        for j in xrange(100000):
            data = [(i, 1) if random.random()>0.8 else (i,0) for i in range(1000)]
            label = [it[1] for it in data]
            yield data, label

    return data_reader
```

```
     input = paddle.layer.data(
                name="input vector", type=paddle.data_type.dense_vector(dim=1000))
    label = paddle.layer.data(
                name="label vector", type=paddle.data_type.dense_vector(dim=1000))
    h1 = paddle.layer.fc(name='hidden1', input=input,
                            size=hidden_size[0],
                            act=paddle.activation.Softmax(),
                            )

    h2 = paddle.layer.fc(name='hidden2', input=h1,
                            size=hidden_size[1],
                            act=paddle.activation.Softmax(),
                            )


    h3 = paddle.layer.fc(name='hidden3', input=h2,
                            size=hidden_size[2],
                            act=paddle.activation.Softmax(),
                            bias_attr=paddle.attr.Param(learning_rate=2),
                            )


    h4 = paddle.layer.fc(name='hidden4', input=h3,
                            size=hidden_size[0],
                            act=paddle.activation.Softmax(),
                            bias_attr=paddle.attr.Param(learning_rate=2),
                            )


    output = paddle.layer.fc(name='output', input=h4,
                            size=input.size,
                            act=paddle.activation.Softmax(),
                            bias_attr=paddle.attr.Param(learning_rate=2))

    cost = paddle.layer.classification_cost(input=output, label=label)
    parameters = paddle.parameters.create(cost)
```

error:
```
I0314 16:06:32.076370 18080 Util.cpp:166] commandline:  --use_gpu=False --trainer_count=1
I0314 16:06:32.166133 18080 GradientMachine.cpp:85] Initing parameters..
I0314 16:06:32.583199 18080 GradientMachine.cpp:92] Init parameters done.
F0314 16:06:32.643702 18080 Matrix.cpp:2417] Check failed: a_col == b_row (2000 vs. 1000)
*** Check failure stack trace: ***
    @     0x7fce1f7f627d  google::LogMessage::Fail()
    @     0x7fce1f7f9d2c  google::LogMessage::SendToLog()
    @     0x7fce1f7f5da3  google::LogMessage::Flush()
    @     0x7fce1f7fb23e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fce1f72568e  paddle::CpuMatrix::mul()
    @     0x7fce1f731e99  paddle::CpuMatrix::mul()
    @     0x7fce1f54859a  paddle::FullyConnectedLayer::forward()
    @     0x7fce1f5b6562  paddle::NeuralNetwork::forward()
    @     0x7fce1f5aa4c3  paddle::GradientMachine::forwardBackward()
    @     0x7fce1f7d21c4  GradientMachine::forwardBackward()
    @     0x7fce1f47f1d9  _wrap_GradientMachine_forwardBackward
    @           0x4b4cb9  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b6c52  PyEval_EvalCode
    @           0x4e1c7d  PyRun_FileExFlags
    @           0x4e3501  PyRun_SimpleFileExFlags
    @           0x4159dd  Py_Main
    @     0x7fce26794bd5  __libc_start_main
    @           0x414b71  (unknown)
    @              (nil)  (unknown)
已放弃
```

why  `a_col == b_row (2000 vs. 1000)` ?
thank you

**Comment by dzhwinter** [comment]
Please double check the `hidden_size` shape, what's the value of it?

For the fc layer, please refer to https://github.com/PaddlePaddle/models/blob/develop/ctr/network_conf.py
, dense_vector works well with fc layer.

**Comment by adrianhust** [comment]
@dzhwinter  hidden_size = [1024, 512, 1024]

**Comment by dzhwinter** [comment]
let's simplify the model,  say if we only have two layers. 
the data shape is [1000, 2], label shape is [1000, 1],

How about
```cpp
data =  densevector(2), 
label = densevector(1)
```
?

**Comment by dzhwinter** [comment]
Please refer to the doc example, I am much more familiar with fluid instead of the old data structure. I highly suggest you start with the fluid. Thanks.

**Comment by dzhwinter** [comment]
@qingqing01 @typhoonzero  Can you help this issue? I'm not familiar with the old matrix data type. I thought the `densevector` should works well in this example.

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
